### PR TITLE
ETCD-362: Add hypershift e2e to openshift/etcd

### PR DESCRIPTION
**Pull request by csrwng**
Adds 2 e2e tests:
- hypershift-e2e (Runs regular HyperShift operator e2e tests) 
- hypershift-e2e-conformance (Runs OpenShift conformance tests on a cluster created with HyperShift)

### CODE REVIEW

**Comment by tjungblu** [comment]
/retitle ETCD-362: Add hypershift e2e to openshift/etcd

File: ci-operator/config/openshift/etcd/openshift-etcd-openshift-4.14.yaml [hunk 1]
```diff
@@ -66,6 +79,14 @@ tests:
   steps:
     cluster_profile: aws
     workflow: openshift-e2e-aws-etcd-recovery
+- as: e2e-hypershift
```
Comment (enxebre) [inline ci-operator/config/openshift/etcd/openshift-etcd-openshift-4.14.yaml:82]: I'd make this required

**Comment by Elbehery** [comment]
@tjungblu do we need this for 4.15 as well ?

File: ci-operator/config/openshift/etcd/openshift-etcd-openshift-4.14.yaml (hunk 1 above)
Comment (tjungblu) [reply ci-operator/config/openshift/etcd/openshift-etcd-openshift-4.14.yaml:82]: I'm fine, if they'd pass. But they currently are not, is this known?

**Comment by tjungblu** [comment]
let's merge this, so we can continue on the rebase, we can debate later around marking those are required

/pj-rehearse ack

**Comment by tjungblu** [comment]
I think there's a silly conflict with the bot changing the golang version.
We're currently talking to DPP in https://redhat-internal.slack.com/archives/CBN38N3MW/p1683806251595059

**Comment by tjungblu** [comment]
/retest

/hold cancel

registry bot was disabled for openshift/etcd and the golang 1.19 bump got merged once more: https://github.com/openshift/release/pull/39318

that should resolve the test conflict against HEAD

**Comment by csrwng** [comment]
/pj-rehearse pull-ci-openshift-etcd-openshift-4.14-e2e-hypershift

**Comment by tjungblu** [comment]
passed :)

/pj-rehearse ack

**Pull request merged**